### DISCUSSION TITLE
Alignment between two containers

### DISCUSSION THREAD

**Discussion by JordanSussman** [terrastruct/d2#1581] [Q&A]
Hello! I'm trying to get the `region a` and `region b` containers within the following code to align at the same level:

```
region a: Primary Region {
  vhub: VWAN HUB {
    firewall: Azure Firewall {
      shape: image
      icon: https://icons.terrastruct.com/azure%2FNetworking%20Service%20Color%2FAzure%20Firewall.svg
    }
  }

  special_vnet: Special VNET {
    VNF1
    VNF2
  }

  p_group: primary group {
    mgmt_vnet: MGMT VNET {
      shape: image
      icon: https://icons.terrastruct.com/azure%2FNetworking%20Service%20Color%2FVirtual%20Networks.svg
    }
    prod_vnet: Prod VNET {
      shape: image
      icon: https://icons.terrastruct.com/azure%2FNetworking%20Service%20Color%2FVirtual%20Networks.svg
    }
  }

  s_group: secondary group {
    mgmt_vnet: MGMT VNET {
      shape: image
      icon: https://icons.terrastruct.com/azure%2FNetworking%20Service%20Color%2FVirtual%20Networks.svg
    }
    prod_vnet: Prod VNET {
      shape: image
      icon: https://icons.terrastruct.com/azure%2FNetworking%20Service%20Color%2FVirtual%20Networks.svg
    }
  }

  vhub <-> p_group
  vhub <-> s_group
  vhub <-> special_vnet.VNF1: BGP peer
  vhub <-> special_vnet.VNF2: BGP peer
}

region b: DR Region {
  vhub: VWAN HUB {
    firewall: Azure Firewall {
      shape: image
      icon: https://icons.terrastruct.com/azure%2FNetworking%20Service%20Color%2FAzure%20Firewall.svg
    }
  }

  special_vnet: Special VNET {
    VNF1
    VNF2
  }

  p_group: Primary group {
    mgmt_vnet: MGMT VNET {
      shape: image
      icon: https://icons.terrastruct.com/azure%2FNetworking%20Service%20Color%2FVirtual%20Networks.svg
    }
    prod_vnet: Prod VNET {
      shape: image
      icon: https://icons.terrastruct.com/azure%2FNetworking%20Service%20Color%2FVirtual%20Networks.svg
    }
  }

  s_group: Secondary group {
    mgmt_vnet: MGMT VNET {
      shape: image
      icon: https://icons.terrastruct.com/azure%2FNetworking%20Service%20Color%2FVirtual%20Networks.svg
    }
    prod_vnet: Prod VNET {
      shape: image
      icon: https://icons.terrastruct.com/azure%2FNetworking%20Service%20Color%2FVirtual%20Networks.svg
    }
  }

  vhub <-> p_group
  vhub <-> s_group
  vhub <-> special_vnet.VNF1: BGP peer
  vhub <-> special_vnet.VNF2: BGP peer
}

region a.vhub.firewall <-> region b.vhub.firewall
```

The aforementioned code results in the two region containers being slightly askew with either the layout of `elk` or `dagre`, see the following screenshot for an example

![image](https://github.com/terrastruct/d2/assets/8935518/54873272-f296-470c-a985-3487839d3d29)

Is it possible to force that the `region a` and `region b` containers are at the same level? I would ideally have the `region b` container to the immediate right of `region a`.

**Comment by ralphtq** [comment]
Is the solution simply to add "direction: right" at the start of the code?

Or will you need to use grids?

**Reply by CanRau** [reply]
If I understood the question correctly, this wouldn't get the desired result I think?

https://play.d2lang.com/?l=&script=7JRNT7NAEMfvfIq5cHwoD8eNMWlV6qWEtBWPzRZG2AjsZnahUdPvboDF0MR4sonRHkiY_87bDswvE4SpEbJmQCIvjOMQ5kLWwBnEJCpOL7AelDcHoC2aPYPkcR7B_cOilwCeBOGBlyWD-WtDCKG17TGALrhCBqLiOVpJpF3Jwhil2WzWWdozSMS1oSY1XiqrGe-yuUEYoTlIehZ17gb-BqkVKbqBfyNLSW4Qzgc3fyzr6Tbvqxyd7nEAtMJU8HLX1mgYbAYLkuhua1tMovD_-BKMUWqXk2wUA2Xn0Js2osorY_OtlqvtNNm575sIMg0v3cC3jvrkxgCKZGZ7i0lmP6i34WuMc9WYyjq7TPbbJtttJ1z9ux5_3amkP5Ema-F1K8BgsYxBIdJXbsHE7fjBiz2D2_WfR0V8QcVZULG5oOL3oIJ7XYA3sqCPHClyeuS8BwAA__8%3D&layout=elk&sketch=0&

**Comment by bo-ku-ra** [comment]
the red line becomes only a straight line.
https://play.d2lang.com/?script=7FZda9swFH33rzgU_FjH06MYg3RbupeG0HTZY1HlO0VUtsSVnJKV_Pfh2M4SVvaWMVhebN_jcz849j3IOFtJXF3hNQMM2-pae9fWTZQQGRDT1lHhg9I2bSXKLAOYjPUNlMSCba14i_se6UoAm3X7JLH6Np3jy9ebAQS-W6YX5ZzE9EfLhNkQHwhAXKtAErZWhg6g1b6RWKcUopxMuigWiZhVTNzqVGhfT1RXMRezOaUXz8-2Mbkol8QbqykX5UfvPOdiNu1p5di6iBsz9Nll_XV_i4G0Ve5x01CSWPYRVvPPD4dhV_PZu1-P4jg7PBr2bZAIgzr78JBZmzoNle9u7x5Oy55fg5Xl1CqXi3Igxt9UAAL7aphxwb76B2ccv9SodSTtm-qi9lnV7jYb768_jL_4KRjfBI9WqeiWRuLmdoFAxH8mihPi7sh3niQ-3V8s503LWVws569ZzvJiOf-L5aiiSypGP9lnj250-kritT8yxcT-mSSYql22y34GAAD__w%3D%3D&layout=elk&
